error/s2n_errno.c

```diff
@@ -371,7 +371,7 @@ int s2n_calculate_stacktrace(void)
     }
 
     int old_errno = errno;
-    GUARD(s2n_free_stacktrace());
+    POSIX_GUARD(s2n_free_stacktrace());
```

Comment:
Why change just this GUARD to POSIX_GUARD?

Reply:
The `errno` module is a little tricky to migrate correctly and there was only a single `GUARD` in here so I figured it would be fine to migrate it along with these changes.